File: fixtures/scale/python/large/cipherlab/checks/check_laddercode.py
from cipherlab.ciphers.laddercode import laddercode_encode, laddercode_decode
from cipherlab.specs.laddercode_spec import laddercode_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_laddercode():
    spec = laddercode_spec()
    for text in corpus_codec():
        packed = laddercode_encode(text)
        if laddercode_decode(packed) != text:
            return False
    return spec["category"] == "codec"

File: fixtures/scale/ruby/large/lib/check_laddercode.rb
require_relative 'laddercode_encode'
require_relative 'laddercode_decode'
require_relative 'corpus_codec'

def check_laddercode
  corpus_codec.each do |text|
    packed = laddercode_encode(text)
    return false if laddercode_decode(packed) != text
  end
  true
end
